core/src/main/java/tc/oc/pgm/stats/StatsModule.java

```diff
@@ -0,0 +1,56 @@
+package tc.oc.pgm.stats;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+import net.kyori.adventure.text.Component;
+import org.bukkit.Color;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.util.math.Formula;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.XMLFluentParser;
+import tc.oc.pgm.util.xml.XMLUtils;
+
+public class StatsModule implements MapModule<StatsMatchModule> {
+  private final List<StatType.OfFormula> formulaStats;
+
+  private StatsModule(List<StatType.OfFormula> formulaStats) {
+    this.formulaStats = formulaStats;
+  }
+
+  @Override
+  public @Nullable StatsMatchModule createMatchModule(Match match) throws ModuleLoadException {
+    return new StatsMatchModule(match, formulaStats);
+  }
+
+  public static class Factory implements MapModuleFactory<StatsModule> {
+    @Override
+    public StatsModule parse(MapFactory factory, Logger logger, Document doc)
+        throws InvalidXMLException {
+      XMLFluentParser parser = factory.getParser();
+      List<Element> statElements = XMLUtils.flattenElements(doc.getRootElement(), "stats");
+      if (statElements.isEmpty()) {
+        return new StatsModule(Collections.emptyList());
+      } else {
+        List<StatType.OfFormula> formulaStats = new ArrayList<>();
+        for (Element el : statElements) {
```

Comment:
Generally the way to use flatten elements is more something like this:
```suggestion
      ImmutableList.Builder<StatType.OfFormula> formulaStats = ImmutableList.builder();
      for (Element el : XMLUtils.flattenElements(doc.getRootElement(), "stats", "stat")) {
```
Which avoids the extra nesting, if it's empty then just formulaStats will be empty at the end.

Also it is good practice to use immutable collections, so you can rest assured that some match won't later modify/break map objects (it would mean if the same map plays again, it may do so with different, and broken, behavior)

Reply:
Cool seems sensible